### PR TITLE
feat: OAT: Rework the JSON editing experience

### PR DESCRIPTION
**Pull request by msnyder-msft**
### Summary of changes 🔍 
Fixing the JSON editor to properly update and validate changes.

### Testing 🧪
- Edit a model in the JSON editor
- Edit a relationship in the JSON editor

### CODE REVIEW

File: src/Components/OATPropertyEditor/Internal/EditorJsonTab/Internal/JSONEditor/JSONEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+    isDTDLReference
+} from '../../../../../../Models/Services/DtdlUtils';
+import {
+    getModelById,
+    getModelIndexById,
+    getReferenceIndexByName
+} from '../../../../../../Models/Context/OatPageContext/OatPageContextUtils';
+import { ensureIsArray } from '../../../../../../Models/Services/OatUtils';
+
+const debugLogging = true;
```
Comment (montgomp) [inline src/Components/OATPropertyEditor/Internal/EditorJsonTab/Internal/JSONEditor/JSONEditor.tsx:49]: Do we want this true?

Comment (msnyder-msft) [reply src/Components/OATPropertyEditor/Internal/EditorJsonTab/Internal/JSONEditor/JSONEditor.tsx:49]: alsfuioasldfuialsekrhyalseuy

Comment (msnyder-msft) [reply src/Components/OATPropertyEditor/Internal/EditorJsonTab/Internal/JSONEditor/JSONEditor.tsx]: Fixed.

**Review by montgomp — APPROVED** [review]
LGTM

**Pull request merged**